File: spec/lua/test_process_coerce_switch.lua
local luaunit = require("luaunit")

require("process_coerce_switch")

TestProcessCoerceSwitch = {}

function TestProcessCoerceSwitch:test_process_coerce_switch()
    local r = ProcessCoerceSwitch:from_file("src/process_coerce_switch.bin")

    luaunit.assertEquals(r.buf_type, 0)
    luaunit.assertEquals(r.flag, 0)
    luaunit.assertEquals(r.buf.bar, "AAAA")
end

File: spec/python/test_process_coerce_switch.py
import unittest

from process_coerce_switch import ProcessCoerceSwitch

class TestProcessCoerceSwitch(unittest.TestCase):
    def test_process_coerce_switch(self):
        r = ProcessCoerceSwitch.from_file("src/process_coerce_switch.bin")

        self.assertEqual(r.buf_type, 0)
        self.assertEqual(r.flag, 0)
        self.assertEqual(r.buf.bar, b"AAAA")
